tests/listsAndDropDowns.spec.ts

```diff
@@ -0,0 +1,84 @@
+import { test, expect } from "@playwright/test";
+
+test.beforeEach(async ({ page }) => {
+  await page.goto("/");
+  await page.getByText("Owners").click();
+  await page.getByText("Search").click();
+  await expect(page.locator("h2")).toHaveText("Owners");
+});
+
+test.describe("Lists and Drop Downs", async () => {
+  test("Validate selected pet types from the list", async ({ page }) => {
+    await page.getByText("George Franklin").click();
+
+    await expect(page.locator(".ownerFullName")).toHaveText("George Franklin");
+
+    await page.getByRole("button", { name: "Edit Pet" }).nth(0).click();
```

Comment:
This locator selects the first "Edit Pet" button, but you need to select a button for the pet with the name "Leo"
If the order of the pets on the page is changed - the wrong pet will be selected.

Please refactor this locator accordingly.

Reply:
done